Stats: Include a post nudge on last post summary components on sites with no posts

youknowriad opened the pull request:
This should fix #5290 

The following nudge is displayed on sites with no posts.

<img width="731" alt="capture d ecran 2016-05-09 a 19 22 40" src="https://cloud.githubusercontent.com/assets/272444/15121935/10de418c-161d-11e6-8e8b-dd1b5b537ffa.png">

**Testing instructions**
The component can have different states : 
 - The site have no posts but we're refreshing the query results => in this case the text is hidden
 - The site have no posts and the query is not requesting => we display the nudge
 - The site have posts => we display the stats and the post title

cc @timmyc

timmyc commented on client/my-sites/stats/post-performance/index.jsx:
```diff
@@ -70,14 +73,12 @@ const StatsPostPerformance = React.createClass( {
-			'is-loading': loading,
-			'is-hidden': ! loading && ! post
-		} );
+		const cardClass = classNames( 'stats-module', 'stats__latest-post-summary', 'is-site-overview' );
```
I think retaining the loading state/class here is still worthwhile to show that a network request is in-progress.

youknowriad replied:
the two classes were only used to hide content on loading state, I replaced those with "ifs" to make a code less confusing.

timmyc commented:
Left a comment that I think we do need to maintain the loading state - see the following screen grabs:

__Loading site with no posts__
![no-post](https://cloud.githubusercontent.com/assets/22080/15130684/3f99cbc2-15ff-11e6-9156-1da1c295ac9a.gif)

__Loading site with post__
![has-post](https://cloud.githubusercontent.com/assets/22080/15130689/47e96a3a-15ff-11e6-86ce-2cd8c158b4c1.gif)

/cc @jancavan for a design review as well.

youknowriad commented:
@timmyc Tks for the review, I'll take a look at how to keep the loading state until the post stat finished the initial loading.

youknowriad commented:
@timmyc I've reuse the StatsModulePlaceholder component while loading (this displays a spinner after some milliseconds) to stay consistent with all the stats UI. Is this ok ?

jancavan commented on client/my-sites/stats/post-performance/index.jsx:
```diff
@@ -112,12 +113,31 @@ const StatsPostPerformance = React.createClass( {
+								<p className="is-empty-message">
+									{ this.translate( 'You haven\'t published any posts yet.' ) }
+								</p>
+								<p>
```
Is it possible to use a <button> here since it's an action and looks like a button?

youknowriad replied:
I'm sorry, it's certainly my "english", but I've not understood what you mean ?

jancavan commented:
This looks good. One minor comment: there's too much padding above the button and too little below it. Something like this would be good:

![screenshot-2016-05-11-09 45 59](https://cloud.githubusercontent.com/assets/4924246/15189022/75babf72-175d-11e6-96a8-4ac314441fb7.jpg)

Should we also add an event to this button?

youknowriad commented:
@jancavan I've just fixed the padding.

<img width="736" alt="capture d ecran 2016-05-11 a 19 20 55" src="https://cloud.githubusercontent.com/assets/272444/15189996/fd957f40-17ad-11e6-8f52-c76aba2245b1.png">

jancavan commented on client/my-sites/stats/post-performance/index.jsx:
```diff
@@ -112,12 +113,31 @@ const StatsPostPerformance = React.createClass( {
+									{ this.translate( 'You haven\'t published any posts yet.' ) }
+								</p>
+								<p>
+									<a href={ newPostUrl } className="button is-primary">
```
Sorry, Github stripped out the word `<button>`.

Is it possible to use `<button>` here instead of `<a>` since it's an action and not just a link?

youknowriad replied:
Actually, It's is just a link. There's nothing done here apart going to the "new post" page.

timmyc commented on client/my-sites/stats/post-performance/index.jsx:
```diff
@@ -70,14 +74,13 @@ const StatsPostPerformance = React.createClass( {
+		const cardClass = classNames( 'stats-module', 'stats-post-performance', 'is-site-overview' );
 
-		const summaryUrl = post ? '/stats/post/' + post.ID + '/' + this.props.site.slug : '#';
+		const newPostUrl = site ? '/post/' + site.ID : '/post';
```
we should use the `site.slug` for this URL.

timmyc commented on client/my-sites/stats/post-performance/index.jsx:
```diff
@@ -112,12 +116,31 @@ const StatsPostPerformance = React.createClass( {
+									{ this.translate( 'You haven\'t published any posts yet.' ) }
+								</p>
+								<div className="stats-post-performance__start-post">
+									<a href={ newPostUrl } className="button is-primary">
```
I think @jancavan had brought this up in a prior comment, but perhaps we can just utilize the [button component](https://github.com/Automattic/wp-calypso/tree/master/client/components/button) here.

timmyc commented:
@youknowriad couple more minor tweaks noted above, but this is functionally working great for me. If we could address those two items and do a rebase then I'd love to get this merged in.

jancavan commented:
Are we adding an event to this button @timmyc?

timmyc commented:
@jancavan as in dropping some tracks or GA events?  Might be valuable to do so.

jancavan commented:
@timmyc yup, a Tracks event would be great.

youknowriad commented:
@timmyc I've just updated the PR. About the `Button` I thought @jancavan was talking about the html `button` tag instead of the component. my bad. I've also added a track event. 

Let me know if there's anything else :)

jancavan commented on client/my-sites/stats/post-performance/style.scss:
```diff
@@ -0,0 +1,7 @@
+}
+
+.stats-post-performance__no-posts-message {
+	margin-bottom: 1em;
```
Is there a reason why this is using `em` and the other one is in `px`? Perhaps just use `px` for both for consistency? I think `14px` will do.

youknowriad replied:
Actually, there is a reason. This overrides a parent "em" margin and I found that using "ems" keeps the spacing good no matter the font-size of the message. 

That said, I can still use px if you prefer :).

jancavan replied:
Well, you could retain `em` if need be, but I wasn't sure why it's two different units instead of just using one or the other.

youknowriad replied:
Just upated to `px` to get a consensus here :)

youknowriad commented:
@timmyc I took all the remarks into account. I rebased. I think the tests are failing because github is experimenting some issues right now. I would love to see this merged :)

jancavan commented:
Design LGTM 👍

timmyc commented:
LGTM too.  Thanks again @youknowriad
